src/transformers/trainer.py

```diff
@@ -2522,10 +2543,16 @@ def evaluation_loop(
             all_preds = nested_truncate(all_preds, num_samples)
         if all_labels is not None:
             all_labels = nested_truncate(all_labels, num_samples)
+        if all_inputs is not None:
+            all_inputs = nested_truncate(all_inputs, num_samples)
 
         # Metrics!
         if self.compute_metrics is not None and all_preds is not None and all_labels is not None:
-            metrics = self.compute_metrics(EvalPrediction(predictions=all_preds, label_ids=all_labels))
+            if args.include_inputs_for_metrics:
+                metrics = self.compute_metrics(EvalPrediction(inputs=all_inputs, predictions=all_preds,
+                                                              label_ids=all_labels))
```

Comment:
```suggestion
            if args.include_inputs_for_metrics:
                metrics = self.compute_metrics(EvalPrediction(predictions=all_preds,
                                                              label_ids=all_labels, inputs=all_inputs))
```
Needs to be last I think.